### ISSUE TITLE
[Feature request] - Insertion of New Tab outside of locked tree

### ISSUE THREAD

**Issue by SHHSSH** [piroor/tst-lock-tree-collapsed#26]
Hi @piroor 

I am writing to request a feature for an issue that is often run into on my behalf.
When I have a locked tree with relevant sub-tabs to the parent-tab to keep organised but temporarily not using them, I often run into an issue where since I am utilising the parent tab often and want to open a New tab via hotkey, the new tab is forced to be a child tab of the parent tab but is not visible or automatically focused due to the tree being locked.

I understand this is relevant to TST's setting of; When a tab is opened from existing tab, open it as - First Child of the parent tab
This is obviously my preferred setting universally and is more intuitive in general.
However, whenever I attempt to open a New tab from a locked tree, which is quite common from my active parent tabs that I'm using, I run into this issue where the child tab is obviously hidden within the locked tree.

Is a feature possible for Lock as collapsed helper addon where a New tab is forced to be independent if the tree is either locked & or collapsed?

This is specifically relevant to an opened New tab(for me personally), rather than an opened link from the parent tab, whereas that may still be more suiting to be opened as a sub-tab of the parent tab/locked tree. However, the latter as a togglable option may still be utilised by some.

As always, thanks in advance for your consideration.

**Comment by piroor** [comment]
![image](https://github.com/piroor/tst-lock-tree-collapsed/assets/70062/7ea787f5-1706-4a4b-bf00-d9fa95d1bf10)
OK I've implemented the option. It depends on new features of TST 4.0 (the planned next version) so please wait until the update.

**Comment by SHHSSH** [comment]
Has this been integrated yet @piroor ? Just spent half an hour fiddling with the new settings in LAC and TST, updated, restarted etc. nothing seems to liberate it from the collapsed tree.

TST Config - https://file.io/RsWuuCTQc06i
LAC Config - https://ibb.co/8PrkXQj

Or maybe it's just some new TST setting I've looked over? I thought maybe it had to do with the one of the latest release notes "Introduce new expert option to expose hidden tabs by other addons, and add expert options UI for the number of pre-rendered pages and visibility of hidden tabs." but I am not sure.

***EDIT*** I've actually got it working with these settings, however the issue is that I believe it's only _visually_ appearing as outside of the locked tree, and when the user Ctrl+Tabs it will not even focus that newly opened blank tab, instead it skips over it as if it was still apart of the tree which is due to a feature I have enabled which helps Ctrl+Tab past locked trees(two settings I believe one in LAC & one in TST).

So basically the option is moot as the newly opened blank/new tab is not functional as it would be generally, nor is it even semi-practically functional as you must expand the locked tree to even access it.

Thank you for everything you do.

**Comment by SHHSSH** [comment]
I believe this could be better achieved with something like "Outdent tab" in TST More Tree Commands.

**Comment by piroor** [comment]
I've realized that the new option of this addon does not work as expected when a new tab is opened by clicking of the "+" button. New tab opened by Ctrl-T (configured to be opened as the first child of the active tab, on TST's options page) is redirected as expected. Did you use the "+" button?

**Comment by piroor** [comment]
I've pushed a change 04d27b3 which fixes the "+" button case. And it is possibly effective on other triggers to open new tabs.

**Comment by SHHSSH** [comment]
I use the Ctrl+T hotkey, not the + button.

**Comment by piroor** [comment]
Hmm, the logic of the released version looks fragile and may not work on some environments. With the recent changes, this addon should detect "newly opened and immediately attached to a collapsed tree" case more robustly. Could you try the development build with about:debugging => "load temporary addon"?
https://github.com/piroor/tst-lock-tree-collapsed/actions/runs/8455849483

**Comment by SHHSSH** [comment]
> Hmm, the logic of the released version looks fragile and may not work on some environments. With the recent changes, this addon should detect "newly opened and immediately attached to a collapsed tree" case more robustly. Could you try the development build with about:debugging => "load temporary addon"? https://github.com/piroor/tst-lock-tree-collapsed/actions/runs/8455849483

Haha, oh mate, you must think I have some form of computer science capability at all/whatsoever. You would be mistaken, lol.
I am very sorry for that and I will do my best to attempt to do so now.

**Comment by SHHSSH** [comment]
I downloaded the xpi file, attempted to install, requires verification; failed.
After about ~5 ChatGPT inquiries, I have found the about:config relevant to such - xpinstall.signatures.required; success.
Yes this seems to work as intended.

What must I do now? Just wait for the upcoming releases?

Thank you so much!

**Comment by piroor** [comment]
Sorry for too less description. Usage of the development build is same to TST's one and here is instructions:
https://github.com/piroor/treestyletab?tab=readme-ov-file#development-builds

**Comment by SHHSSH** [comment]
@piroor I dislike that I have to keep xpinstall.signatures.required - false 
I find it a potential security risk, but am using it temporarily. In the future will I need not worry about this after the developer edition syncs with the public release?

***EDIT/HELP*** I've always wondered piroor, if you can select a tab via sifting through them with Ctrl+Tab/Ctrl+Shift+Tab but not have the tab load? I never sought the answer years ago thinking perhaps it tiresome for me to make another "Feature request" or something.

Is that at all possible? It would do wonders for me, one who often sifts through tabs swiftly with mouse macros. I ofcourse would then have to adjust to the functionality by utilising a macro that actually loads the tab when I want to.

**I'm not actually requesting this, I don't want you to go out of your way, I just wonder if it's currently possible.**

If at all relevant;
simulateTabsLoadInBackgroundInverted - unchecked
Keep tabs pending (unloaded) when tabs are accidentaly focused - checked
Avoid pending (unloaded) tabs to be activated accidentally on current tab closes or tree collapsing - checked

**Comment by piroor** [comment]
>  I find it a potential security risk, but am using it temporarily. In the future will I need not worry about this after the developer edition syncs with the public release?

The development build is just a testing purpose. I'll release it as a new version soon, and you won't need to deactivate the signature check after it is released.

> if you can select a tab via sifting through them with Ctrl+Tab/Ctrl+Shift+Tab but not have the tab load?

Did you mean that A) you are finding a way that Ctrl-Tab/Ctrl-Shift-Tab actually activate all tabs whether it is pending or not but keep it pending while the Ctrl key is pressed? Or, B) Ctrl-Tab/Ctrl-Shift-Tab should skip pending tabs but currently they are activated and loaded unintentionally?

If you meant A, sorry but there is no option to do that.
If you meant B, your settings should work as that except some special cases: when you start Ctrl-Tab/Ctrl-Shift-Tab on about: pages or Mozilla Add-ons website, Ctrl-Tab/Ctrl-Shift-Tab won't skip pending tabs - this is due to a restriction of WebExtensions API.

**Comment by piroor** [comment]
I've released 1.4.3. It will be downloadable soon.

**Comment by SHHSSH** [comment]
> > I find it a potential security risk, but am using it temporarily. In the future will I need not worry about this after the developer edition syncs with the public release?
> 
> The development build is just a testing purpose. I'll release it as a new version soon, and you won't need to deactivate the signature check after it is released.
> 
> > if you can select a tab via sifting through them with Ctrl+Tab/Ctrl+Shift+Tab but not have the tab load?
> 
> Did you mean that A) you are finding a way that Ctrl-Tab/Ctrl-Shift-Tab actually activate all tabs whether it is pending or not but keep it pending while the Ctrl key is pressed? Or, B) Ctrl-Tab/Ctrl-Shift-Tab should skip pending tabs but currently they are activated and loaded unintentionally?
> 
> If you meant A, sorry but there is no option to do that. If you meant B, your settings should work as that except some special cases: when you start Ctrl-Tab/Ctrl-Shift-Tab on about: pages or Mozilla Add-ons website, Ctrl-Tab/Ctrl-Shift-Tab won't skip pending tabs - this is due to a restriction of WebExtensions API.

Oh no. I really was just talking about cycling through tabs with Ctrl+Tab and have it _**not**_ actually "Load" the "Unloaded" tab.
It's no worries mate, don't worry about it :)
